core/src/test/scala/unit/kafka/log/LogValidatorTest.scala

```diff
@@ -773,6 +773,23 @@ class LogValidatorTest {
       partitionLeaderEpoch = RecordBatch.NO_PARTITION_LEADER_EPOCH)
   }
 
+  @Test(expected = classOf[InvalidRecordKeyException])
+  def testNullKeyForCompactedTopic() {
```

Comment:
maybe also add a test that verifies the comment here: https://github.com/apache/kafka/pull/2875/files#diff-c0425b7649a53be33a7bc71842896299R306